presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java

```diff
@@ -73,4 +78,9 @@ public Connector create(String catalogName, Map<String, String> requiredConfig,
 
         return injector.getInstance(JdbcConnector.class);
     }
+
+    public interface JdbcModuleProvider
+    {
+        Module getModule(String catalogName);
```

Comment:
Alternatively, we could bind `@CatalogName String` in guice and let it be injected in other places.

Reply:
I considered this but we need it at binding time for JMX exports. Otherwise this would be a nicer approach.